src/software/simulation/physics/physics_world.h

```diff
@@ -136,11 +136,18 @@ class PhysicsWorld
     void stepSimulation(const Duration& time_step);
 
     /**
-     * Returns the friendly PhysicsRobots currently in the world
+     * Returns the yellow PhysicsRobots currently in the world
      *
-     * @return the friendly PhysicsRobots in the world
+     * @return the yellow PhysicsRobots in the world
      */
-    std::vector<std::weak_ptr<PhysicsRobot>> getFriendlyPhysicsRobots() const;
+    std::vector<std::weak_ptr<PhysicsRobot>> getYellowPhysicsRobots() const;
+
+    /**
+     * Returns the blue PhysicsRobots currently in the world
+     *
+     * @return the blue PhysicsRobots in the world
+     */
+    std::vector<std::weak_ptr<PhysicsRobot>> getBluePhysicsRobots() const;
```

Comment:
why are we using weak ptr?

Reply:
Ideally we don't expose the "internals" of the physics world because it has a very explicit order of destruction internally (due to Box2D internals). However we have to in order for the `SimulatorRobots` and singletons to actually control robots in the physics environment.

We expose `weak_ptrs` so anything outside of the physics world gets a non-owning reference, meaning that a consumer can't accidentally extend the lifetime of the physics object beyond the PhysicsWorld and cause issues on destruction.